src/Subscription.js

```diff
@@ -52,6 +45,24 @@ class Subscription extends EventEmitter {
         if (this.resendOptions.from == null && this.resendOptions.to != null) {
             throw new Error('"from" must be defined as well if "to" is defined.')
         }
+        this.initialResendDone = Object.keys(this.resendOptions).length === 0
+        this.state = Subscription.State.unsubscribed
+        this.groupKeys = groupKeys || {}
+        this.queue = []
```

Comment:
- Messages in the initial resend should be processed as soon as possible (correct order is reached), instead of buffering the whole resend into memory and starting to process messages when the resend is complete
- Would be nice to avoid having the queuing and queue processing code in this class

Could the `OrderingUtil` be used for the initial resend as well?

What if `orderingUtil.add(msg, forceQueue)` takes a second parameter which is used to force the realtime messages into the queue until the initial resend is complete?